src/controllers/v2/renewal.js

```diff
@@ -0,0 +1,171 @@
+      return {status: 404, message: `Registration ${registrationNumber} not allocated.`};
+    }
+
+    const existingReg = findOneregistrationResponse;
+
+    const mappedExistingReg = {
+      convictions: existingReg.convictions,
+      usingGL01: existingReg.usingGL01,
+      usingGL02: existingReg.usingGL02,
+      usingGL03: existingReg.usingGL03,
+      complyWithTerms: existingReg.complyWithTerms,
+      meatBaits: existingReg.meatBaits,
+      fullName: existingReg.fullName,
+      addressLine1: existingReg.addressLine1,
+      addressLine2: existingReg.addressLine2,
+      addressTown: existingReg.addressTown,
+      addressCounty: existingReg.addressCounty,
+      addressPostcode: existingReg.addressPostcode,
+      phoneNumber: existingReg.phoneNumber,
+      emailAddress: existingReg.emailAddress,
+      uprn: existingReg.uprn
+    };
+
+    // Clean up the user's input before we store it in the database.
+    const cleanObject = cleanInput(request.body);
+
+    const mappedCleanObject = {
+      convictions: cleanObject.convictions,
+      usingGL01: cleanObject.usingGL01,
+      usingGL02: cleanObject.usingGL02,
+      usingGL03: cleanObject.usingGL03,
+      complyWithTerms: cleanObject.complyWithTerms,
+      meatBaits: cleanObject.meatBaits,
+      fullName: cleanObject.fullName,
+      addressLine1: cleanObject.addressLine1,
+      addressLine2: cleanObject.addressLine2,
+      addressTown: cleanObject.addressTown,
+      addressCounty: cleanObject.addressCounty,
+      addressPostcode: cleanObject.addressPostcode,
+      phoneNumber: cleanObject.phoneNumber,
+      emailAddress: cleanObject.emailAddress,
+      uprn: cleanObject.uprn
+    };
+
+    const changes = {};
+
+    // Check if there are any changes from the original
+    for (const key in mappedExistingReg) {
+      if (mappedExistingReg[key] !== mappedCleanObject[key]) {
+        changes[key] = mappedCleanObject[key];
+      }
+    }
+
+    await db.sequelize.transaction(async (t) => {
+      try {
+        if (Object.keys(changes).length > 0) {
+          await RegistrationHistory.create(
+            {
+              ...mappedExistingReg,
+              RegistrationId: registrationNumber,
+              createdByLicensingOfficer: existingReg.createdByLicensingOfficer,
+              // Need to confirm expected behaviour here!
+              expiryDate: existingReg.expiryDate
+            },
+            {transaction: t}
+          );
+
+          await Registration.update(changes, {where: {id: registrationNumber}, transaction: t});
+        }
+
+        // For now, renewals expiry date will be null 12/11/2024
+        await Renewal.create({RegistrationId: registrationNumber, expiryDate: null}, {transaction: t});
+      } catch (error) {
+        console.log(error);
+        return {status: 500, message: `Error occured: ${error}`};
+      }
+    });
+
+    return {status: 201, message: `Registration ${registrationNumber} renewed.`};
```

Comment:
I think we want to return the registration number itself so it can be displayed on the success page, rather than a message and a string.

Reply:
Not sure, as we pass the same registration number in as a path parameter of the endpoint